src/openrct2/ride/Ride.cpp

```diff
@@ -3729,14 +3729,14 @@ money32 set_operating_setting_nested(ride_id_t rideId, RideSetSetting setting, u
  *
  *  rct2: 0x006B4CC1
  */
-static int32_t ride_mode_check_valid_station_numbers(Ride* ride)
+static StationIndex ride_mode_check_valid_station_numbers(Ride* ride)
 {
-    uint8_t no_stations = 0;
-    for (uint8_t station_index = 0; station_index < MAX_STATIONS; ++station_index)
+    uint8_t numStations = 0;
```

Comment:
Seems likely to blow up in the future, I'd just use a large primitive instead.
```suggestion
    uint32_t numStations = 0;
```

Reply:
`uint16_t` should more than enough - that allows for 65535 stations _per ride_. (I'm not sure we're going to increase the maximum amount of stations beyond 255, but we're _certainly_ not going to increase it beyond 65535...)

Reply:
Alright.